easybuild/tools/options.py

```diff
@@ -1047,8 +1047,50 @@ def _postprocess_checks(self):
 
         self.log.info("Checks on configuration options passed")
 
+    def get_cfg_opt_abs_path(self, opt_name, path):
+        """Get path value of configuration option as absolute path."""
+        if os.path.isabs(path):
+            abs_path = path
+        else:
+            abs_path = os.path.abspath(path)
+            self.log.info("Relative path value for '%s' configuration option resolved to absolute path: %s",
+                          opt_name, abs_path)
+        return abs_path
+
+    def _ensure_abs_path(self, opt_name):
+        """Ensure that path value for specified configuration option is an absolute path."""
+
+        opt_val = getattr(self.options, opt_name)
+        if opt_val:
+            if isinstance(opt_val, string_type):
+                setattr(self.options, opt_name, self.get_cfg_opt_abs_path(opt_name, opt_val))
+            elif isinstance(opt_val, list):
+                abs_paths = [self.get_cfg_opt_abs_path(opt_name, p) for p in opt_val]
+                setattr(self.options, opt_name, abs_paths)
+            else:
+                error_msg = "Don't know how to ensure absolute path(s) for '%s' configuration option (value type: %s)"
+                raise EasyBuildError(error_msg, opt_name, type(opt_val))
+
     def _postprocess_config(self):
         """Postprocessing of configuration options"""
+
+        # resolve relative paths for configuration options that specify a location
+        path_opt_names = ['buildpath', 'containerpath', 'git_working_dirs_path', 'installpath',
+                          'installpath_modules', 'installpath_software', 'prefix', 'packagepath',
+                          'robot', 'robot_paths', 'sourcepath']
```

Comment:
Drop robot and return the "[os.path.abspath(path) for path in self.options.robot" part on (new) line 1136

Reply:
Any reason why?
It's more consistent like this, and the absolute path resolution is done earlier, so less change for a relative path to cause trouble early on...

Reply:
Oh, because of the failing test, we're now unintentionally promoting an argument like `example.eb` to an absolute path if it's used directly after `--robot`, which messes up the easyconfig searching...

And people wonder why testing is so important. :P

Reply:
fixed in 6934dd8